set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java

```diff
@@ -374,17 +388,85 @@ public void addNewItemToList(UserSet existingUserSet, int positionInt, String ne
 	}
 
     @Override
-    public UserSet updateUserSetsWithIsDefinedByUrl(UserSet storedUserSet, String apiKey, String action)
+    public UserSet updateUserSetsWithIsDefinedByUrl(UserSet storedUserSet, String apiKey, String action,
+    		String sort, String sortOrder, int pageNr, int pageSize)
 	    throws HttpException, IOException, JSONException {
     	
     	String uri;
-    	uri = storedUserSet.getIsDefinedBy();
+    	String additionalParameters;
+    	additionalParameters = buildSearchQuery(sort, sortOrder, pageNr, pageSize);
+    	uri = storedUserSet.getIsDefinedBy() + additionalParameters;
     	SearchApiResponse apiResult = getSetApiService().queryEuropeanaApi(uri, apiKey, action);
     	List<String> items = new ArrayList<String>();
     	for(String item : apiResult.getItems()) {
     		items.add(WebUserSetFields.BASE_URL_DATA + item);
     	}
-    	storedUserSet.setItems(items);
+    	if (items.size() > 0) {
+	    	storedUserSet.setItems(items);
+	    	storedUserSet.setTotal(items.size());
+    	}
     	return storedUserSet;
     }
+    
+	/**
+	 * This method appends additional search parameter from HTTP request
+	 * @param sort
+	 * @param sortOrder
+	 * @param pageNr
+	 * @param pageSize
+	 * @return additional search Query string
+	 */
+	public String buildSearchQuery(String sort, String sortOrder, int pageNr, int pageSize) {
+
+		StringBuilder searchQuery = new StringBuilder();
+		
+		searchQuery.append(WebUserSetFields.AND);
+		searchQuery.append(pageNr);
+		searchQuery.append(WebUserSetFields.PAGE).append("=");
+		if(pageNr < 0)
+			searchQuery.append(WebUserSetFields.DEFAULT_PAGE);
+		else
+			searchQuery.append(pageNr);
+
+		searchQuery.append(WebUserSetFields.AND);
+		searchQuery.append(WebUserSetFields.PAGE_SIZE).append("=");
+		if(pageSize < 0)
+			searchQuery.append(WebUserSetFields.MAX_ITEMS_PER_PAGE);
+		else
+			searchQuery.append(pageSize);
+		
+		searchQuery.append(WebUserSetFields.AND);
+		if (!Strings.isNullOrEmpty(sort)) {
+			searchQuery.append(WebUserSetFields.PARAM_SORT).append("=");
+			searchQuery.append(sort);
+			searchQuery.append(WebUserSetFields.AND);
+			searchQuery.append(WebUserSetFields.PARAM_SORT_ORDER).append("=");
+			searchQuery.append(sortOrder);
+		}
+
+		return searchQuery.toString();
+	}
+        
+    /* (non-Javadoc)
+     * @see eu.europeana.set.web.service.UserSetService#updateUserSetsWithCloseSetItems(eu.europeana.set.definitions.model.UserSet, java.util.List)
+     */
+    public UserSet updateUserSetsWithCloseSetItems(UserSet storedUserSet, List<String> items) {    	
```

Comment:
is this method really needed? It only sets 2 values in the userSet

Reply:
I believe, it is required to support application logic for open/closed sets according to documentation